typescript/agentkit/src/wallet-providers/privyEvmEmbeddedWalletProvider.ts

```diff
@@ -0,0 +1,517 @@
+      method: "personal_sign",
+      params: {
+        message,
+        encoding: "utf-8",
+      },
+    };
+
+    try {
+      const response = await this.executePrivyRequest<PrivyResponse<{ signature: Hex }>>(body);
+      return response.data?.signature;
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Message signing failed: ${error.message}`);
+      }
+      throw new Error("Message signing failed");
+    }
+  }
+
+  /**
+   * Signs typed data according to EIP-712.
+   *
+   * @param typedData - The typed data object to sign
+   * @param typedData.domain - The domain object containing contract and chain information
+   * @param typedData.types - The type definitions for the structured data
+   * @param typedData.primaryType - The primary type being signed
+   * @param typedData.message - The actual data to sign
+   * @returns A Address that resolves to the signed typed data as a hex string
+   */
+  async signTypedData(typedData: {
+    domain: Record<string, unknown>;
+    types: Record<string, Array<{ name: string; type: string }>>;
+    primaryType: string;
+    message: Record<string, unknown>;
+  }): Promise<Hex> {
+    const body = {
+      address: this.#address,
+      chain_type: "ethereum",
+      chain_id: NETWORK_ID_TO_CHAIN_ID[this.#network.chainId!],
+      ...typedData,
+    };
+
+    try {
+      const response = await this.executePrivyRequest<{ signature: Hex }>({
+        method: "eth_signTypedData_v4",
+        params: body,
+      });
+      return response.signature;
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error("Typed data signing failed: " + error.message);
+      }
+      throw new Error("Typed data signing failed with unknown error");
+    }
+  }
+
+  /**
+   * Signs a transaction.
+   *
+   * @param transaction - The transaction to sign.
+   * @returns The signed transaction.
+   */
+  async signTransaction(transaction: TransactionRequest): Promise<Hex> {
+    const body = {
+      address: this.#address,
+      chain_type: "ethereum",
+      method: "eth_signTransaction",
+      params: {
+        transaction: {
+          ...transaction,
+          from: this.#address,
+        },
+      },
+    };
+
+    try {
+      const response =
+        await this.executePrivyRequest<PrivyResponse<{ signed_transaction: Hex }>>(body);
+      return response.data?.signed_transaction;
+    } catch (error) {
+      if (error instanceof Error) {
+        throw new Error(`Transaction signing failed: ${error.message}`);
+      }
+      throw new Error("Transaction signing failed");
+    }
+  }
+
+  /**
+   * Sends a transaction.
+   *
+   * @param transaction - The transaction to send.
+   * @returns The hash of the transaction.
+   */
+  async sendTransaction(transaction: TransactionRequest): Promise<Hex> {
+    const body = {
+      address: this.#address,
+      chain_type: "ethereum",
+      method: "eth_sendTransaction",
```

Comment:
To get sendTransaction to work, I had to add 
```caip2: `eip155:${this.#network.chainId!}`, ```

to the body.

Easiest way to test is via a USDC transfer
1. `Get me USDC from the faucet`
2. `You can use the ERC20 actions related to transfer. Use the ERC20 action "transfer" used for ERC20 tokens. In the schema, use an amount of 500000, a contractAddress of 0x036CbD53842c5426634e7929541eC2318f3dCF7e and a destination of 0x799f3da72Dd622f77aC8fFDfDC97808468f2784D`

Reply:
true. fixing now